src/main/java/com/solace/spring/cloud/stream/binder/config/autoconfigure/JCSMPSessionConfiguration.java

```diff
@@ -0,0 +1,133 @@
+package com.solace.spring.cloud.stream.binder.config.autoconfigure;
+
+import com.solace.spring.cloud.stream.binder.config.SolaceBinderClientInfoProvider;
+import com.solace.spring.cloud.stream.binder.config.SolaceHealthIndicatorsConfiguration;
+import com.solace.spring.cloud.stream.binder.health.contributors.SolaceBinderHealthContributor;
+import com.solace.spring.cloud.stream.binder.health.handlers.SolaceSessionEventHandler;
+import com.solace.spring.cloud.stream.binder.health.indicators.SessionHealthIndicator;
+import com.solace.spring.cloud.stream.binder.oauth.SolaceSessionOAuth2TokenProvider;
+import com.solace.spring.cloud.stream.binder.oauth.SpringJCSMPFactory;
+import com.solace.spring.cloud.stream.binder.provisioning.SolaceEndpointProvisioner;
+import com.solace.spring.cloud.stream.binder.util.JCSMPSessionEventHandler;
+import com.solacesystems.jcsmp.Context;
+import com.solacesystems.jcsmp.ContextProperties;
+import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.JCSMPSession;
+import com.solacesystems.jcsmp.impl.JCSMPBasicSession;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import java.io.ByteArrayOutputStream;
+import java.util.*;
+
+import static com.solacesystems.jcsmp.XMLMessage.Outcome.*;
+
+@Slf4j
+@RequiredArgsConstructor
+@Configuration
+@Import(SolaceHealthIndicatorsConfiguration.class)
+@ConditionalOnProperty(name = "solace.java.host")
+public class JCSMPSessionConfiguration {
+    private final static Map<String, SessionCacheEntry> SESSION_CACHE = new HashMap<>();
+
+    @Bean
+    public JCSMPSessionEventHandler jcsmpSessionEventHandler(JCSMPProperties jcsmpProperties,
+                                                             Optional<SolaceBinderHealthContributor> sessionHealthIndicator,
+                                                             Optional<SolaceSessionEventHandler> solaceSessionEventHandler,
+                                                             Optional<SolaceSessionOAuth2TokenProvider> solaceSessionOAuth2TokenProvider) {
+        return ensureSessionCache(jcsmpProperties, sessionHealthIndicator, solaceSessionEventHandler, solaceSessionOAuth2TokenProvider).jcsmpSessionEventHandler();
+    }
+
+    @Bean
+    public JCSMPSession jcsmpSession(JCSMPProperties jcsmpProperties,
```

Comment:
Does this still fits the requirement of the feature:
JcsmpSession needs to be reused from leader election lib?

Reply:
yes